unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/NFTPromptHUD/NFTPromptHUDView.cs

```diff
@@ -0,0 +1,306 @@
+﻿using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+using DCL;
+using DCL.Helpers;
+using DCL.Helpers.NFT;
+using System.Linq;
+using DCL.Interface;
+
+public class NFTPromptHUDView : MonoBehaviour
+{
+    [SerializeField] internal GameObject content;
+
+    [SerializeField] RawImage imageNft;
+    [SerializeField] Image imageNftBackground;
+    [SerializeField] TextMeshProUGUI textNftName;
+    [SerializeField] TextMeshProUGUI textOwner;
+
+    [Header("Last Sale")]
+    [SerializeField] TextMeshProUGUI textLastSaleSymbol;
+    [SerializeField] TextMeshProUGUI textLastSalePrice;
+    [SerializeField] TextMeshProUGUI textLastSaleNeverSold;
+
+    [Header("Price")]
+    [SerializeField] TextMeshProUGUI textPriceSymbol;
+    [SerializeField] TextMeshProUGUI textPrice;
+    [SerializeField] TextMeshProUGUI textPriceNotForSale;
+
+    [Header("Description & Comment")]
+    [SerializeField] TextMeshProUGUI textDescription;
+    [SerializeField] TextMeshProUGUI textComment;
+    [SerializeField] GameObject containerDescription;
+    [SerializeField] GameObject containerComment;
+
+    [Header("Spinners")]
+    [SerializeField] GameObject spinnerGeneral;
+    [SerializeField] GameObject spinnerNftImage;
+
+    [Header("Buttons")]
+    [SerializeField] internal Button buttonClose;
+    [SerializeField] internal Button buttonCancel;
+    [SerializeField] internal Button buttonOpenMarket;
+    [SerializeField] TextMeshProUGUI textOpenMarketButton;
+
+    Coroutine fetchNFTRoutine = null;
+    Coroutine fetchNFTImageRoutine = null;
+    IWrappedTextureAsset imageAsset = null;
+
+    bool backgroundColorSet = false;
+    string marketUrl = null;
+
+    private void Awake()
+    {
+        buttonClose.onClick.AddListener(Hide);
+        buttonCancel.onClick.AddListener(Hide);
+        buttonOpenMarket.onClick.AddListener(OpenMarketUrl);
```

Comment:
It would be a good practice to unsubscribe these subscriptions somewhere along the monobehaviour life-cycle (in this case in the OnDestroy method).

Reply:
In this case, buttons being children of the GameObject with that MonoBehavior attached, If the GameObject is destroyed that means the buttons will be destroyed as well... so I don't think unsubscribing to events is really necessary

Reply:
Ok! Then there is no problem! Approved ;-)